supplementary/constraintsolver/src/GSolver.cpp

```diff
@@ -357,6 +362,12 @@ void GSolver::rPropLoop(const autodiff::Tape& tape, const double* seed, const st
     } else {
         initialPoint(tape, o_result, limits, curGradient);
     }
+    for (int i = 1; i <= dim; ++i) {
```

Comment:
can use std::any_of?

Reply:
Need to use a plain for loop here because o_result is not a proper c++ iteratable with begin() and end() but a pointer to double.

Reply:
you mean `curGradient` is not an iterable, right?

Reply:
starting at i=1 is correct?

Reply:
Start at 1 is correct. See similar existing loops below.
As far as I understood the code i=0 is the optimization result and 1,2,3 are the current inputs.